upstairs/src/lib.rs

```diff
@@ -2140,76 +2085,75 @@ struct BackpressureConfig {
  * These methods are how to add or checking for new work on the Guest struct
  */
 impl Guest {
-    pub fn new(log: Option<Logger>) -> Guest {
+    pub fn new(log: Option<Logger>) -> (Guest, GuestIoHandle) {
         let log = log.unwrap_or_else(build_logger);
-        Guest {
-            /*
-             * Incoming I/O requests are added to this queue.
-             */
-            reqs: Mutex::new(VecDeque::new()),
-            notify: Notify::new(),
-            /*
-             * The active hashmap is for in-flight I/O operations
-             * that we have taken off the incoming queue, but we have not
-             * received the response from downstairs.
-             * Note that a single IO from outside may have multiple I/O
-             * requests that need to finish before we can complete that IO.
-             */
-            guest_work: Mutex::new(GuestWork {
+        let (req_tx, req_rx) = mpsc::channel(500);
```

Comment:
It'd be worth adding a comment for how this number was chosen, even if it's arbitrary

Reply:
Added a longer comment in 1be982b7173d7bdf9d3485e02dfc7cad08bbd46e